tui-markdown/src/style_sheet.rs

```diff
@@ -0,0 +1,89 @@
+//! Style sheet abstraction for tui-markdown.
+//!
+//! The library used to hard–code all colour and attribute choices in an
+//! internal `styles` module.  That made it impossible for downstream crates to
+//! provide their own look-and-feel without forking the crate.  This file
+//! introduces the [`StyleSheet`] trait: a minimal interface that supplies the
+//! handful of [`ratatui::style::Style`] values the renderer needs.
+//!
+//! Users that are happy with the stock colours do not have to do anything –
+//! the crate exports a [`DefaultStyleSheet`] which matches the old behaviour
+//! and is used by default.  Projects that want to theme the output can
+//! implement the trait for their own type and pass an instance via
+//! [`crate::Options`].
+
+use ratatui::style::{Color, Modifier, Style};
+
+/// A collection of `ratatui::style::Style`s consumed by the renderer.
+///
+/// The trait purposefully stays tiny: whenever the renderer needs a colour
+/// choice we add a new getter here.  The default implementation maintains full
+/// backward-compatibility with the styles that lived in the old `mod styles`.
+pub trait StyleSheet: Clone + Send + Sync + 'static {
+    /// Style for a Markdown heading.  `level` is one-based (`1` for `# H1`, …).
+    fn heading(&self, level: u8) -> Style;
+
+    /// Style for inline `code` spans and fenced code blocks when syntax
+    /// highlighting is disabled.
+    fn code(&self) -> Style;
+
+    /// Style for the text of an inline or reference link.
+    fn link(&self) -> Style;
+
+    /// Base style applied to blockquotes (`>` prefix and body text).
+    fn blockquote(&self) -> Style;
+}
+
+/* ------------------------------------------------------------------------- */
+/*  Built-in style sheet – identical to the previous hard-coded colours       */
+/* ------------------------------------------------------------------------- */
```

Comment:
nit: I prefer to avoid borders in code like this. Fold it into the doc comments.